File: exercise_1/template/Solution2.m
function Solution2(input_file_name)

%% read from txt
%  (either use the following code, or any other read_txt method you like)
input_table = readtable(input_file_name, 'ReadVariableNames',false);
input_matrix = table2cell(input_table);


%% Enter your code here.
%  Find the solution for matrix in input_filename and print the output in 'output_for_task2.txt' file
% ...
%  Assume your output matrix is of type cell


%% convert cell to table
%  write the table to a txt file
output_table = cell2table(output_matrix);
writetable(board_table, 'output_for_task2.txt', 'Delimiter',' ', 'WriteVariableNames', false);


end

File: exercise_1/template/Solution2.java
import java.io.*;
import java.util.*;

public class Solution2 {

    static char[][] solveTask2(char[][] board) {
        /*
        Enter your code here
        Return the updated board
         */
        return board;
    }

    public static void main(String[] args) {
        try {
            /* Read the input matrix*/
            char[][] input = readFile(args[0]);

            /* Your main function to solve the problem*/
            char[][] output = solveTask2(input);

            /*Print the solution matrix to a txt file*/
            printMatrix(output, "output_for_task2.txt");

        } catch (Exception e) {
            System.out.println(e.getMessage());
        }



    }

    public static void printMatrix(char[][]board, String filepath){
        try (PrintWriter writer = new PrintWriter(filepath, "UTF-8")) {
            for(int i =0;i<board.length;i++){
                for(int j= 0;j<board[i].length;j++){
                    writer.print(board[i][j]);
		    if(j < board[i].length - 1){
			writer.print(" ");
		    }
                }
                writer.println();
            }
            writer.close();
        } catch (Exception e) {
            e.printStackTrace();
        }

    }
    public static char[][] readFile(String filepath) throws FileNotFoundException {
        ArrayList<ArrayList<String>> array = new ArrayList<>();
        Scanner input = new Scanner(new File(filepath));
        while(input.hasNextLine())
        {
            Scanner colReader = new Scanner(input.nextLine());
            ArrayList col = new ArrayList();
            while(colReader.hasNext("[.X]"))
            {
                col.add(colReader.next());
            }
            if(col.size() > 0){
            	array.add(col);
	    }
        }
        input.close();

        char[][] inputMatrix = new char[array.size()][array.get(0).size()];
        for(int i=0;i<array.size();i++) {
            for(int j=0;j<array.get(0).size();j++){
                inputMatrix[i][j] = array.get(i).get(j).toCharArray()[0];
            }
        }
        return inputMatrix;
    }
}
